apps/meteor/app/ui-message/client/actionButtons/lib/applyButtonFilters.ts

```diff
@@ -16,14 +16,16 @@ import {
 
 import { hasAtLeastOnePermission, hasPermission, hasRole, hasAnyRole } from '../../../../authorization/client';
 
-export const applyAuthFilter = (button: IUIActionButton, room?: IRoom): boolean => {
+export const applyAuthFilter = (button: IUIActionButton, room?: IRoom, ignoreSubscriptions = false): boolean => {
 	const { hasAllPermissions, hasOnePermission, hasAllRoles, hasOneRole } = button.when || {};
 
 	const userId = Meteor.userId();
 
 	const hasAllPermissionsResult = hasAllPermissions ? hasPermission(hasAllPermissions) : true;
 	const hasOnePermissionResult = hasOnePermission ? hasAtLeastOnePermission(hasOnePermission) : true;
-	const hasAllRolesResult = hasAllRoles ? !!userId && hasAllRoles.every((role) => hasRole(userId, role, room?._id)) : true;
+	const hasAllRolesResult = hasAllRoles
+		? !!userId && hasAllRoles.every((role) => hasRole(userId, role, room?._id, ignoreSubscriptions))
+		: true;
 	const hasOneRoleResult = hasOneRole ? !!userId && hasAnyRole(userId, hasOneRole, room?._id) : true;
```

Comment:
Maybe this validation also requires the `ignoreSubscriptions` argument to be passed?

Reply:
make sense, adding